lib/plugins/lindat_auth/__init__.py

```diff
@@ -1,163 +1,242 @@
+        A Shibboleth authentication module with a failover
+        solution. Please note that the failover solution should be
+        used only in exceptional cases because no high security
+        measures have been applied (as opposed to federated login).
     """
+    ID_KEYS = ('HTTP_EPPN', 'HTTP_PERSISTENT_ID', 'HTTP_MAIL')
+    RESERVED_USER = '__user_count'
 
-    MIN_PASSWORD_LENGTH = 5
+    def get_user_info(self, user_id):
+        raise NotImplementedError()
 
-    def __init__(self, db_conn, sessions, corplist, admins, anonymous_id):
+    def __init__(self, public_corplist, db, sessions, conf, failover):
         """
-        arguments:
-        db_conn -- a database connection
-        sessions -- a session plugin
-        corplist -- a list of permitted corpora
-        admins -- list of usernames with administrator privileges
-        anonymous_id -- an ID of a public/anonymous user
-        """
-        super(LINDATAuth, self).__init__(anonymous_id=anonymous_id)
-        self.db_conn = db_conn
-        self.sessions = sessions
-        self.corplist = corplist
-        self.admins = admins
 
-    def anonymous_user(self):
-        return dict(id=0, user=None, fullname=_('anonymous'))
+        Arguments:
+            public_corplist -- default public corpora list
+            db_provider -- default database
+            sessions -- a session plugin
+
+        Note:
+            login_url - used e.g., in dialog ``
+        """
+        anonymous_id = int(conf['anonymous_user_id'])
+        super(FederatedAuthWithFailover, self).__init__(anonymous_id=anonymous_id)
+        self._db = db
+        self._sessions = sessions
+        self._public_corplist = public_corplist
+        self._failover_auth = failover
+        self._logout_url = conf['logout_url']
+        self._conf = conf
 
     def validate_user(self, plugin_api, username, password):
         """
-        returns:
-        a dict with user properties or empty dict
+            Try to find the user using two methods.
         """
-        getenv = plugin_api.get_environ
-        if username is not None and username != '':
-            cols = ('id', 'user', 'pass', 'firstName', 'surname')
-            cursor = self.db_conn.cursor()
-            try:
-                cursor.execute("SELECT %s FROM user WHERE user = %%s" % ','.join(cols), (username, ))
-                row = cursor.fetchone()
-                if row and crypt.crypt(password, row[2]) == row[2]:
-                    row = dict(zip(cols, row))
-                else:
-                    row = {}
-                cursor.close()
-                if 'id' in row:
-                    return dict(id=row['id'],
-                                user=row['user'],
-                                fullname='%s %s' % (row['firstName'], row['surname']))
-            except:
+        if username is not None and 0 < len(username):
+            if username in FederatedAuthWithFailover.RESERVED_USER:
```

Comment:
RESERVED_USER is `'__user_count'` L43 either I'm missing something or this is not what one would expect

Reply:
if a user would be named `__user_count` it would do strange things

Reply:
ok, I get it; though the variable name is slightly confusing; somehow I think about system lvl accounts (administrator etc) when I see "RESERVED_USER". 
Why use `in` though and not `==`?
```
>>> 'count' in '__user_count'
True
```
Might not be a problem in this case, but we don't mind substrings, or do we? And with `in` I'm kind of expecting a set (list, dict...) to follow...